modules/ossm-installing-gateway-using-gateway-injection.adoc

```diff
@@ -0,0 +1,246 @@
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f gateway-deployment.yml
+----
+
+. Verify that the gateway `Deployment` rollout was successful by running the following command:
++
+[source,terminal]
+----
+$ oc rollout status deployment/<gateway_name> -n <gateway_namespace>
+----
+
+. Create a YAML file named `gateway-service.yml` that contains the Kubernetes `Service` for the gateway. 
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: <gateway_name>
+  namespace: <gateway_namespace>
+spec:
+  type: ClusterIP <1>
+  selector:
+    istio: <gateway_name> <2>
+  ports:
+    - name: status-port
+      port: 15021
+      protocol: TCP
+      targetPort: 15021
+    - name: http2
+      port: 80
+      protocol: TCP
+      targetPort: 80
+    - name: https
+      port: 443
+      protocol: TCP
+      targetPort: 443
+----
+<1> When you  set `spec.type` to `ClusterIP` the gateway `Service` is available to traffic within the cluster. If the gateway handles ingress traffic from outside the cluster, set `spec.type` to `LoadBalancer`. Alternatively, you can use OpenShift Routes. 
+<2> Set the `selector` to the unique label or set of labels specified in the pod template of the gateway `Deployment` that you previously created.
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f gateway-service.yml
+----
+
+. Verify that the gateway `Service` is targeting the `Endpoints` of gateway pods by running the following command:
++
+[source,terminal]
+----
+$ oc get endpoints <gateway_name> -n <gateway_namespace>
+----
++
+You should see output similar to the following example:
++
+.Example output
+[source,terminal]
+----
+NAME              ENDPOINTS                             AGE
+<gateway_name>    10.131.0.181:8080,10.131.0.181:8443   1m
+----
+
+. Optional: Create a YAML fle named `secret-reader.yml` that defines a `HorizontalPodAutoscaler` for the gateway. The following example sets the minimum replicas to `2` and the maximum replicas to `5` and scales the replicas up when average CPU utilization exceeds 80% of the CPU resource limit. This limit is specified in the pod template of the `Deployment` for the gateway.  
++
+[source,yaml,subs="attributes,verbatim"]
+----
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: <gateway_name>
+  namespace: <gateway_namespace>
+spec:
+  minReplicas: 2
+  maxReplicas: 5
+  metrics:
+  - resource:
+      name: cpu
+      target:
+        averageUtilization: 80
+        type: Utilization
+    type: Resource
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: <gateway_name> <1>
+----
+<1> Set `spec.scaleTargetRef.name` to the name of the gateway `Deployment` previously created.
+
+. Optional: Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f gateway-hpa.yml
+----
```

Comment:
Just a thought: There's quite a few of these pairs of steps that detail creating a file and then applying that file to the cluster. Could or should each pair be combined into a single step?

Reply:
discussed in slack and decided to keep as is.